### DISCUSSION TITLE
vc env improvement

### DISCUSSION THREAD

**Discussion by azezsan** [vercel/vercel#8983] [Ideas]
I'm working on a sveltekit project and since sveltekit uses vite under the hood the env variables should be prefixed with VITE_ to be exposed, while vercel knows that this is a sveltekit project and it does add the VITE_ prefix at build time it would be very convenient if it were to add them automatically when running `vercel env pull`  currently I have to run it then it overrides my previous envs and I have to go again and add it myself every time I use the CLI

if it's a detection limitation from the CLI at least give me the option to choose a custom prefix

**Comment by swarnava** [comment]
Can you share us a screencast? I don't think I have understood the problem correctly. Thanks!

**Comment by azezsan** [comment]
I just did `vercel env pull`
[image: image.png]

What I'm saying is this output isn't very helpful, because I still have to
add `VITE_` prefix to every variable like this

[image: image.png]

now since I already linked my project as seen below

[image: image.png]

and the CLI knows this is a sveltekit project. why doesn't it add the
prefix for me? because it's annoying to go and add it every time I want to
use `vercel env pull` because it keeps overwriting it with the useless
output above, vercel also knows that this is required at build time and it
does that as shown here
https://vercel.com/docs/concepts/projects/environment-variables#system-environment-variables,
so why does it help me during development?

if this was a detection limitation error it should at least give me the
option to add a default custom prefix
